knative-operator/pkg/common/service_monitor.go

```diff
@@ -2,27 +2,53 @@ package common
 
 import (
 	"context"
+	"fmt"
+	"os"
 
-	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis"
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	mfclient "github.com/manifestival/controller-runtime-client"
+	mf "github.com/manifestival/manifestival"
 	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
 	kubemetrics "github.com/operator-framework/operator-sdk/pkg/kube-metrics"
 	"github.com/operator-framework/operator-sdk/pkg/metrics"
+	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
+	eventingv1alpha1 "knative.dev/operator/pkg/apis/operator/v1alpha1"
+	kmeta "knative.dev/pkg/kmeta"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func SetupServiceMonitor(ctx context.Context, cfg *rest.Config, metricsPort int32, operatorMetricsPort int32) error {
+const (
+	EventingBrokerServiceMonitorPath     = "deploy/resources/broker-service-monitors.yaml"
+	EventingSourceServiceMonitorPath     = "deploy/resources/source-service-monitor.yaml"
+	EventingSourcePath                   = "deploy/resources/source-service.yaml"
+	SourceLabel                          = "eventing.knative.dev/source"
+	SourceNameLabel                      = "eventing.knative.dev/sourceName"
+	SourceRoleLabel                      = "sources.knative.dev/role"
+	TestEventingBrokerServiceMonitorPath = "TEST_EVENTING_BROKER_SERVICE_MONITOR_PATH"
+	TestMonitor                          = "TEST_MONITOR"
+	TestSourceServiceMonitorPath         = "TEST_SOURCE_SERVICE_MONITOR_PATH"
+	TestSourceServicePath                = "TEST_SOURCE_SERVICE_PATH"
+)
+
+func SetupServerlessOperatorServiceMonitor(ctx context.Context, cfg *rest.Config, api client.Client, metricsPort int32, metricsHost string, operatorMetricsPort int32) error {
 	// Commented below to avoid a stream of these errors at startup:
 	// E1021 22:50:03.372487       1 reflector.go:134] github.com/operator-framework/operator-sdk/pkg/kube-metrics/collector.go:67: Failed to list *unstructured.Unstructured: the server could not find the requested resource
-	// if err = serveCRMetrics(cfg); err != nil {
-	// 	log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
-	// }
+	if err := serveCRMetrics(cfg, metricsHost, operatorMetricsPort); err != nil {
+		log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
+	}
```

Comment:
The enablement of this should've been its own PR. I don't think it necessarily has anything to do with the eventing dashboards.

Reply:
Ok I refactored stuff and improved the existing code. It could have been a separate PR. Next time will do.